### PR TITLE
Various Cleanups, including PREFERRED_VERSION, and vendor/xilinx

### PR DESCRIPTION
**Pull request by mhatle**
This series of changes does the following:

* Moves libmetal-xlnx and open-amp-xlnx to the vendor/xilinx location
* Rename libmetal_dev to libmetal-dev, adding specific python enabling and AUTOREV disable/enable
* Rename open-amp_dev to open-amp-dev, adding specific python enabling and AUTOREV disable/enable
* Rename vendor/xilinx open-amp/libmetal to other names
* Remove obsolete PNBLACKLIST entires from vendor/xilinx
* Rework layer.conf / vendor/xilinx layer.inc interactions
* Fix broken open-amp-xlnx 2022.2 to point to a working REPO.

### CODE REVIEW

File: recipes-openamp/libmetal/libmetal-dev_git.bb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+SRCBRANCH ?= "main"
+SRCREV = "${@oe.utils.conditional("PREFERRED_PROVIDER_libmetal", "libmeta-dev", "${AUTOREV}", "2a59968271231b747bdf8fde0201cefe44aacc8c", d)}"
```
Comment (wmamills) [inline recipes-openamp/libmetal/libmetal-dev_git.bb:2]: libmetal-dev is misspelled as libmeta-dev above

**Comment by bentheredonethat** [comment]
any update on this PR? 

CC @mhatle  @wmamills

**Comment by wmamills** [comment]
The discussion is happening off PR in email.
Mark is going to look at the issue with libmetal-dev and openamp-image-minimal.

I am not likely to have time to look at this again until Dec 1, unless Mark can identify the issue I was having.
The OpenAMP webinar must take higher priority

**Comment by mhatle** [comment]
Working on it today.. hopefully have some answers in a few hours.

**Comment by mhatle** [comment]
Pull request updated.  The issue found has been fixed/corrected.  Due to the PN changing the output package names as changing as well causing INSTALL_IMAGE to not find the specific name it was looking for.  This was corrected using a generate RPROVIDES.

Additionally the LICENSE for the libmetal-dev was incorret and has been fixed, the open-amp-demos package was not being created due to incorrect override syntax.

diff --git a/recipes-openamp/libmetal/libmetal-dev_git.bb b/recipes-openamp/libmetal/libmetal-dev_git.bb
index a3fc569..f1f2a18 100644
--- a/recipes-openamp/libmetal/libmetal-dev_git.bb
+++ b/recipes-openamp/libmetal/libmetal-dev_git.bb
@@ -1,6 +1,7 @@
 SRCBRANCH ?= "main"
 SRCREV = "${@oe.utils.conditional("PREFERRED_PROVIDER_libmetal", "libmetal-dev", "${AUTOREV}", "2a59968271231b747bdf8fde0201cefe
44aacc8c", d)}"
 BRANCH = "main"
+LIC_FILES_CHKSUM ?= "file://LICENSE.md;md5=f4d5df0f12dcea1b1a0124219c0dbab4"
 PV = "${SRCBRANCH}+git${SRCPV}"
 DEFAULT_PREFERENCE = "-1"

diff --git a/recipes-openamp/libmetal/libmetal.inc b/recipes-openamp/libmetal/libmetal.inc
index b0c29e3..90f87e4 100644
--- a/recipes-openamp/libmetal/libmetal.inc
+++ b/recipes-openamp/libmetal/libmetal.inc
@@ -53,10 +53,13 @@ ALLOW_EMPTY:${PN}-demos = "1"
 
 PACKAGES:append = " ${PN}-demos"
 
+RPROVIDES:${PN} += "libmetal"
 FILES:${PN} = " \
     ${libdir}/*.so* \
     ${bindir}/test-* \
 "
+
+RPROVIDES:${PN}-demos += "libmetal-demos"
 FILES:${PN}-demos = " \
     ${bindir}/libmetal_* \
     ${bindir}/*demo \
diff --git a/recipes-openamp/open-amp/open-amp.inc b/recipes-openamp/open-amp/open-amp.inc
index de8e12b..ab261eb 100644
--- a/recipes-openamp/open-amp/open-amp.inc
+++ b/recipes-openamp/open-amp/open-amp.inc
@@ -62,13 +62,15 @@ python rpmsg_no_ipi_setup() {
 # only add definition after toolchain file has been generated
 do_generate_toolchain_file[postfuncs] += "rpmsg_no_ipi_setup"
 
-ALLOW_EMPTY_${PN}-demos = "1"
+ALLOW_EMPTY:${PN}-demos = "1"
 PACKAGES:append = " ${PN}-demos"
 
+RPROVIDES:${PN} += "open-amp"
 FILES:${PN} = " \
     ${libdir}/*.so* \
 "
 
+RPROVIDES:${PN}-demos += "open-amp-demos"
 FILES:${PN}-demos = " \
     ${bindir}/*-shared \
 "

**Review by wmamills — APPROVED** [review]
OK from me

**Comment by bentheredonethat** [comment]
@wmamills  LGTM

**Comment by wmamills** [comment]
@bentheredonethat @mhatle I would greatly prefer Ben's DTSI request to be a new PR.
I am trying to get everything we are showing for the OpenAMP webinar into upstream meta-openamp instead of having to point to my fork.  You guys can work the DTSI at your own pace and it does not matter if that is before or after the demo build.

**Comment by bentheredonethat** [comment]
@wmamills  ok sounds good to me. will open a new PR once this is merged with that.

**Pull request merged**